### PR TITLE
Adds a dev section to the bottom of the dependency tab

### PR DESCRIPTION
**Pull request by zbynek**
Related to issue https://issues.jenkins.io/browse/WEBSITE-386

Adds a dev section to the bottom of the dependency tab:

<details><summary>Plugin in BOM</summary>

image

</details>

<details><summary>Standard plugin</summary>

image
</details>

... and a link to detailed usage stats in the sidebar
<details><summary>Same for all plugins</summary>

image

</details>

CC @daniel-beck

### CODE REVIEW

File: src/components/MavenDependency.jsx [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+            {hasBomEntry && <p>
+                <>
+                    To avoid version conflicts it is suggested not to depend on a specific version, but use the
+                    {' '}
+                    <a target="_blank" href="https://github.com/jenkinsci/bom#readme" rel="noreferrer">Jenkins plugin BOM</a>
+                    {' '}
+                    and the following depndency snippet:
+                </>
+            </p>}
+            <pre>
```
Comment (halkeye) [inline src/components/MavenDependency.jsx]: we have the plugin to make things click to copy if its useful here. I was going to say I don't know who would find the maven coords useful till I saw the jira ticket.

https://github.com/jenkins-infra/plugin-site/blob/ef75f3fc7c80916626db32ca7cc78138a3a58103/src/components/InstallViaCLI.jsx#L8

File: src/templates/plugin.jsx [hunk 2]
```diff
@@ -95,12 +99,13 @@ function PluginPage({data: {jenkinsPlugin: plugin, reverseDependencies: reverseD
                                 installations={plugin.stats.installations}
                             />
                         </div>
+                        <div className="label-link"><a target="_blank" href={`https://stats.jenkins.io/pluginversions/${plugin.name}.html`} rel="noreferrer">Show details</a></div>
                     </div>
                     <div className="sidebarSection">
                         <h5>Links</h5>
-                        {plugin.scm && plugin.scm.link && <div className="label-link"><a href={plugin.scm.link}>GitHub</a></div>}
+                        {plugin.scm && plugin.scm.link && <div className="label-link"><a target="_blank" href={plugin.scm.link} rel="noreferrer">GitHub</a></div>}
```
Comment (halkeye) [inline src/templates/plugin.jsx]: I'm generally not a fan of forcing new tabs to happen, takes away the agency from the user. Is there a request for new tab, or just a preference?

**Comment by halkeye** [comment]
Couple small notes but looks good to me.

**Review by halkeye — APPROVED** [review]

**Comment by zbynek** [comment]
@daniel-beck  any :+1:/:-1: on including the Maven dependency? It was requested in a comment, so it extends the scope of your ticket. Hopefully it's hidden well enough not to bother majority of users and still provides some value for plugin devs.

**Comment by timja** [comment]
I love including the maven dependency, I think I would use it a lot

**Comment by zbynek** [comment]
@halkeye screenshots updated to show the clipboard copy buttons. It's now extracted to a component, so the release tab changed slightly as well:
image

**Comment by halkeye** [comment]
> @halkeye screenshots updated to show the clipboard copy buttons. It's now extracted to a component, so the release tab changed slightly as well:

Thats really awesome. I try to (remember to) make things components as much as possible so its easier to refactor and improve. This is a great improvement

File: src/templates/plugin.jsx [hunk 3]
```diff
@@ -95,6 +99,7 @@ function PluginPage({data: {jenkinsPlugin: plugin, reverseDependencies: reverseD
                                 installations={plugin.stats.installations}
                             />
                         </div>
+                        <div className="label-link"><a href={`https://stats.jenkins.io/pluginversions/${plugin.name}.html`}>Installs by version</a></div>
```
Comment (daniel-beck) [inline src/templates/plugin.jsx]:
```suggestion
                        <div className="label-link"><a href={`https://stats.jenkins.io/pluginversions/${plugin.name}.html`}>View detailed version information</a></div>
```

It's not just _plugin_ versions that are installed, also _core_ -- that may actually be the more important part anyway.

**Comment by daniel-beck** [comment]
Ironically this will be least useful for plugins that have tons of dependencies already, because the bottom of that page is very far down 😄 Perhaps this could be inserted at the top of the "Dependent plugins" section, which is typically the long section for popular dependencies (especially detached plugins), and also makes sense semantically ("How to add a plugin to this section")?

**Comment by zbynek** [comment]
@daniel-beck thanks for the suggestion, I moved the dependencies up.

**Comment by zbynek** [comment]
@daniel-beck OK to merge?

File: src/components/MavenDependency.jsx [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+            <h3>Declaring a dependency</h3>
+            <p>As a plugin developer you can use this plugin as dependency of your plugin by adding a dependency tag to your POM.</p>
+            {!hasBomEntry && <p>To add the latest version of this plugin as a maven dependency, use the following:</p>}
+            {hasBomEntry && <p>
+                <>
+                    To avoid version conflicts it is suggested not to depend on a specific version, but use the
+                    {' '}
+                    <a href="https://github.com/jenkinsci/bom#readme">Jenkins plugin BOM</a>
+                    {' '}
+                    and the following depndency snippet:
```
Comment (daniel-beck) [inline src/components/MavenDependency.jsx]:
```suggestion
                    and the following dependency snippet:
```

**Review by daniel-beck — APPROVED** [review]
Thanks 👍 

Looking forward to seeing it live!

**Pull request merged**